Xamarin.PropertyEditing.Mac/Controls/DateTimeEditorControl.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections;
+using AppKit;
+using Xamarin.PropertyEditing.Mac.Resources;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class DateTimeEditorControl : PropertyEditorControl<PropertyViewModel<DateTime>>
+	{
+		readonly DateTimeTextField editor;
+
+		public DateTimeEditorControl ()
+		{
+			editor = new DateTimeTextField {
+				Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+			};
+
+			// update the value on keypress
+			editor.Changed += (sender, e) => {
+				if (DateTime.TryParse (editor.StringValue, out DateTime dateValue)) {
+					ViewModel.Value = dateValue;
+					UpdateValue ();

```

Comment:
you shouldn't need to call `UpdateValue` here. Assigning `ViewModel.Value` should automagically call `UpdateValue` under the hood.

Reply:
mm... then this hiddes an issue updating viewmodel, because I added because was not refreshing it.

Reply:
If you look at the `StringEditorControl`, or any of the other editors, there are no calls to `UpdateValue` after `ViewModel.Value` is set. So I agree, there must be something else going on that is breaking the correct set of events.